gateway/sds_gateway/users/views.py

```diff
@@ -1893,3 +2408,875 @@ def get(self, request, *args, **kwargs) -> JsonResponse:
+        file_uuid = kwargs.get("uuid")
+        if not file_uuid:
+            return JsonResponse({"error": "File UUID required"}, status=400)
+
+        file_obj = get_object_or_404(File, uuid=file_uuid, is_deleted=False)
+
+        # Access control: owner or shared via capture/dataset
+        has_access = False
+        if file_obj.owner_id == request.user.id:
+            has_access = True
+        elif file_obj.capture_id:
+            has_access = user_has_access_to_item(
+                request.user, str(file_obj.capture_id), ItemType.CAPTURE
+            )
+        elif file_obj.dataset_id:
+            has_access = user_has_access_to_item(
+                request.user, str(file_obj.dataset_id), ItemType.DATASET
+            )
+
+        if not has_access:
+            return JsonResponse({"error": "Not found or access denied"}, status=404)
+
+        try:
+            # Size guard
+            if file_obj.size and int(file_obj.size) > self.MAX_BYTES:
+                return JsonResponse({"error": "File too large to preview"}, status=413)
+
+            # Read content
+            file_obj.file.open("rb")
+            raw = file_obj.file.read(self.MAX_BYTES + 1)
+            file_obj.file.close()
+
+            if len(raw) > self.MAX_BYTES:
+                return JsonResponse({"error": "File too large to preview"}, status=413)
+
+            # Detect JSON by extension
+            name_lower = (file_obj.name or "").lower()
+            if name_lower.endswith(".json"):
+                try:
+                    parsed = json.loads(raw.decode("utf-8"))
+                    pretty = json.dumps(parsed, indent=2, ensure_ascii=False)
+                    return HttpResponse(
+                        pretty, content_type="text/plain; charset=utf-8"
+                    )
+                except (UnicodeDecodeError, json.JSONDecodeError):
+                    # fall through to plain text rendering below
+                    pass
+
+            # Default: return UTF-8 text
+            try:
+                text = raw.decode("utf-8")
+            except UnicodeDecodeError:
+                # Replace undecodable bytes
+                text = raw.decode("utf-8", errors="replace")
+
+            return HttpResponse(text, content_type="text/plain; charset=utf-8")
+        except OSError:
+            logger.exception("Error reading file content for preview")
+            return JsonResponse({"error": "Error reading file"}, status=500)
+
+
+class FileH5InfoView(Auth0LoginRequiredMixin, View):
+    """Return a summarized structure for an HDF5 file as JSON for modal preview."""
+
+    H5_MAX_BYTES = 200 * 1024 * 1024  # 200 MiB cap for preview temp copy
+    MAX_PREVIEW_ELEMENTS = 10
+    MAX_CHILDREN_PER_GROUP = 200
+    MAX_RECURSION_DEPTH = 4
+
+    def get(self, request: HttpRequest, *args, **kwargs) -> JsonResponse:  # noqa: C901, PLR0911, PLR0912, PLR0915
```

Comment:
there are 4 functions defined in this method, maybe we can move them to the top level of this file, or even to a different file?